service/src/main/java/am/smartcafe/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package am.smartcafe.impl;
+
+import am.smartcafe.model.UserChangePassword;
+import am.smartcafe.repository.UserRepository;
+import am.smartcafe.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService{
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private UserRepository userRepository;
+
+
+
+    private static final String REGEX = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z]).{6,20}$";
+
+
+
+    public String changePassword( UserChangePassword userChangePassword){
+        User user = getById(userChangePassword.getUserId());
+        String failText = "Your password must be at least 6 characters," +
+                " and must include at least one upper case letter," +
+                " one lower case letter, and one numeric digit.";
+
+
+        if (passwordEncoder.matches(userChangePassword.getPassword(),user.getPassword()) &&
+        userChangePassword.getRepeatNewPassword().equals(userChangePassword.getNewPassword())
+          && REGEX.matches(userChangePassword.getNewPassword())){
+            String newPassword = passwordEncoder.encode(userChangePassword.getNewPassword());
+            user.setPassword(newPassword);
+            user.setActive(true);
+            userRepository.save(user);
+            return null;
+        }
+        return failText;
+    }
+
+
+
+
+
+    @Bean
+    public PasswordEncoder encoder() {
```

Comment:
would be good to move it into the configuration class so everywhere on bean could be used